app/code/Magento/Newsletter/Model/Plugin/CustomerPlugin.php

```diff
@@ -165,6 +168,38 @@ public function afterGetById(CustomerRepository $subject, CustomerInterface $cus
         return $customer;
     }
 
+    /**
+     * Plugin after getById customer that obtains newsletter subscription status for given customer.
+     *
+     * @param CustomerRepository $subject
+     * @param SearchResults $searchResults
+     * @param SearchCriteria $searchCriteria
+     * @return SearchResults
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function afterGetList(
+        CustomerRepository $subject,
+        SearchResults $searchResults,
+        SearchCriteriaInterface $searchCriteria
+    ) {
+
+        foreach ($searchResults->getItems() as $customer) {
+            $extensionAttributes = $customer->getExtensionAttributes();
```

Comment:
Hi @enriquei4,
This change will definitely work, but it will cause performance issue as it executes retrieving subscription information for each customer separately.
Could you update your code to fetch needed data one query using collection?